src/Featurizers/StandardScaleWrapperFeaturizer.h

```diff
@@ -0,0 +1,331 @@
+template <typename InputT, typename TransformedT, size_t MaxNumTrainingItemsV>
+StandardScaleWrapperEstimator<InputT, TransformedT, MaxNumTrainingItemsV>::StandardScaleWrapperEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex, bool withMean, bool withStd) :
+    BaseType(
+        "StandardScaleWrapperEstimator",
+        pAllColumnAnnotations,
+        [pAllColumnAnnotations, colIndex](void) { return Components::StatisticalMetricsEstimator<InputT, MaxNumTrainingItemsV>(std::move(pAllColumnAnnotations), std::move(colIndex)); },
+        [pAllColumnAnnotations, colIndex](void) { return Components::StandardDeviationEstimator<InputT, MaxNumTrainingItemsV>(std::move(pAllColumnAnnotations), std::move(colIndex)); },
+        [pAllColumnAnnotations, colIndex, &withMean, &withStd](void) { return Details::StandardScaleWrapperEstimatorImpl<InputT, TransformedT, MaxNumTrainingItemsV>(std::move(pAllColumnAnnotations), std::move(colIndex), std::move(withMean), std::move(withStd)); }
+    ) {
+}
+
+// ----------------------------------------------------------------------
+// |
+// |  Details::StandardScaleWrapperEstimatorImpl
+// |
+// ----------------------------------------------------------------------
+template <typename InputT, typename TransformedT, size_t MaxNumTrainingItemsV>
+Details::StandardScaleWrapperEstimatorImpl<InputT, TransformedT, MaxNumTrainingItemsV>::StandardScaleWrapperEstimatorImpl(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex, bool withMean, bool withStd) :
+    BaseType("StandardScaleWrapperEstimatorImpl", std::move(pAllColumnAnnotations)),
+    _colIndex(
+        std::move(
+            [this, &colIndex](void) -> size_t & {
+                if(colIndex >= this->get_column_annotations().size())
+                    throw std::invalid_argument("colIndex");
+
+                return colIndex;
+            }()
+        )
+    ),
+    _withMean(std::move(withMean)),
+    _withStd(std::move(withStd)) {
+}
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+template <typename InputT, typename TransformedT, size_t MaxNumTrainingItemsV>
+bool Details::StandardScaleWrapperEstimatorImpl<InputT, TransformedT, MaxNumTrainingItemsV>::begin_training_impl(void) /*override*/ {
+    return false;
+}
+
+template <typename InputT, typename TransformedT, size_t MaxNumTrainingItemsV>
+FitResult Details::StandardScaleWrapperEstimatorImpl<InputT, TransformedT, MaxNumTrainingItemsV>::fit_impl(typename BaseType::InputType const *, size_t) /*override*/ {
+    throw std::runtime_error("This will never be called");
+}
+
+template <typename InputT, typename TransformedT, size_t MaxNumTrainingItemsV>
+void Details::StandardScaleWrapperEstimatorImpl<InputT, TransformedT, MaxNumTrainingItemsV>::complete_training_impl(void) /*override*/ {
+}
+
+}
+}
+}
```

Comment:
Can you add namespace comments here please.

Reply:
namespace comments are added.